libs/core/src/scss/interaction-state/_layer.scss

```diff
@@ -0,0 +1,81 @@
+@use 'sass:math';
+@use 'sass:meta';
+@use '../utils';
+
+// Put state layer above content layer as default.
+// You can change move the content layer on top by declaring e.g.:
+//
+//    --content-layer-z-index: calc(#{layer.$state-layer-z-index} + 1);
+//
+// in a component that uses interaction state layer.
+$_content-layer-z-index: utils.z('default');
+$state-layer-z-index: $_content-layer-z-index + 1;
```

Comment:
I don't think I agree entirely, but it's good you point it out. We should definitely have a discussion about it (outside of code review 😁). 

In my opinion, the `$z-layers` map is a double-edged sword: It's a good thing to keep related things together, but depending on the stacking contexts in which these values are used you won't get what you'd expect. An element with z-index of 1 _can_ be on top of another element with a z-index of 100 if it's in a context "above" the second one.

This means that you could have a layer where you use e.g. `avatarBadge` (2) and not be able to make e.g. `loading-overlay` (1001) "win" over it, which would be very confusing. It all depends on the stacking contexts and the relation between the different layers.

In this case, it's not relevant what the actual values are, as long as `$state-layer-z-index` has a higher value than `$_content-layer-z-index`. They are not related or dependant on any of the other values in the `$z-layers` map. This is why they are only declared locally.

Reply:
Another thing is, that `$_content-layer-z-index` must be a private member while `$state-layer-z-index` must be public. That's not possible if we use the `$z-layers` map.

Yet another thing 🙂, neither variable should be open for modification. Since `$z-layers` is declared using `!default` I think you'd be able to change the value in the map.

Reply:
Good points! 

Regarding storing the z-index values in a maps, i've been wondering myself if it makes sense due to stacking contexts. So i agree with what you're saying. 

There might have been a _kings tanks_ (google translates version of "kongstanke") with the z-layers that could be interesting to see if we can track down. Otherwise we just have to form it in our own image 👼 

Feel free to resolve this conversation!